src/JJMasterData.Commons/Language/JJMasterDataTranslatorProvider.cs

```diff
@@ -0,0 +1,170 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+using JJMasterData.Commons.Dao;
+using JJMasterData.Commons.Dao.Entity;
+using JJMasterData.Commons.DI;
+using JJMasterData.Commons.Options;
+using Microsoft.Extensions.Localization;
+
+namespace JJMasterData.Commons.Language;
+
+public class JJMasterDataTranslatorProvider : ITranslatorProvider
+{
+    internal IStringLocalizer<JJMasterDataResources> StringLocalizer { get; }
+    internal IEntityRepository EntityRepository { get; }
+
+    public JJMasterDataTranslatorProvider(IEntityRepository entityRepository, IStringLocalizer<JJMasterDataResources> stringLocalizer)
+    {
+        EntityRepository = entityRepository;
+        StringLocalizer = stringLocalizer;
+    }
+
+    public IDictionary<string, string> GetLocalizedStrings(string culture)
+    {
+        string tableName = JJService.Options.ResourcesTableName;
+        
+        if (string.IsNullOrEmpty(tableName))
+            return new Dictionary<string,string>();
+        if (string.IsNullOrEmpty(JJService.Options.GetConnectionString()))
+            return new Dictionary<string,string>();
+
+        var element = GetElement(tableName);
+        if (!EntityRepository.TableExists(element.TableName))
+            EntityRepository.CreateDataModel(element);
+        
+        var stringLocalizerValues = GetStringLocalizerValues();
+        var databaseValues = GetDatabaseValues(element, culture);
+
+        if (databaseValues.Count > 0)
+        {
+            databaseValues.ToList().ForEach(x => stringLocalizerValues[x.Key] = x.Value);
```

Comment:
@LucioPelinson  This solves our problem of truncating the resources table at every change 😁

Reply:
`JJMasterDataLocalizationProvider` has been lost the reference with `DbTranslatorProvider` so I don't know compare!<br>
I think the file was deleted and created instead of renamed.

Reply:
Yeah, I don't know if this happened because it was a LOT of changes or the file was wrongfully renamed by me